From 67c401648fbd51de503d99ce4fca27774638f254 Mon Sep 17 00:00:00 2001
From: Leonard Hecker <leonard@hecker.io>
Date: Tue, 14 Oct 2025 19:53:51 +0200
Subject: [PATCH] Fix Rust nightly builds (#668)

`panic_immediate_abort` is being stablized as `panic = immediate-abort`
(yay!).
See: https://github.com/rust-lang/rust/issues/147286

Closes #657
---
 .cargo/release-nightly.toml | 16 ++++++++++++++++
 README.md                   |  4 +++-
 2 files changed, 19 insertions(+), 1 deletion(-)
 create mode 100644 .cargo/release-nightly.toml

diff --git a/.cargo/release-nightly.toml b/.cargo/release-nightly.toml
new file mode 100644
index 000000000000..f294dca85b6a
--- /dev/null
+++ b/.cargo/release-nightly.toml
@@ -0,0 +1,16 @@
+[profile.release]
+panic = "immediate-abort"
+
+[target.'cfg(all(target_os = "windows", target_env = "msvc"))']
+rustflags = [
+    "-Ctarget-feature=+crt-static",
+    "-Clink-args=/DEFAULTLIB:ucrt.lib",
+    "-Clink-args=/NODEFAULTLIB:vcruntime.lib",
+    "-Clink-args=/NODEFAULTLIB:msvcrt.lib",
+    "-Clink-args=/NODEFAULTLIB:libucrt.lib",
+]
+
+[unstable]
+panic-immediate-abort = true
+build-std = ["std", "panic_abort"]
+build-std-features = ["default", "optimize_for_size"]
